EventsService moved to own DI, no longer part of the PluginsHooksService

jeabakker opened the pull request:
- chore(di): correctly register EventsService

first some other stuff

jeabakker committed:
chore(di): correctly register EventsService

mrclay commented:
My logic was that hooks/events are basically the same thing and most components end up depending on both in the end.

jeabakker committed:
chore(hooks): custom unregister code in correct service

Some PluginHooksService code was present in the
HooksRegistrationService. Since the registration was handled in the
PluginHooksService the unregister part should also be handled there.

jeabakker committed:
chore(events): function documentation
